File: leetCode/fukun/101-200/lc_111.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

const minDepth = function (root) {
    const helper = function (node, depth) {
        if (!node.left && !node.right) return depth;
        return Math.min(node.left ? helper(node.left, depth + 1) : Infinity, node.right ? helper(node.right, depth + 1) : Infinity);
    };
    if (!root) return 0;
    return helper(root, 1);
};

let t = new TreeNode(1);
t.left = new TreeNode(2);
t.right = new TreeNode(2);
console.log(minDepth(t));

File: leetCode/XSS/lc_111.java
public class lc_111 {
    public static void main(String[] args) {

    }
    public static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode(int x) { val = x; }
    }
    static class Solution {
        public int minDepth(TreeNode root) {
            return depth(root,0);
        }
        public int depth(TreeNode r,int index)
        {
            if(r==null)return index;
            else if(r.left==null&&r.right==null) return index+1;
            else if(r.left==null)return depth(r.right,index+1);
            else if(r.right==null)return depth(r.left,index+1);
            else return Math.min(depth(r.left,index+1),depth(r.right,index+1));
        }
    }
}
